lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: n
+# Space Complexity: 1
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  current = 0
+  result = 0
+
+  until list.length - 1 < current 
+    if list[result] != list[current] 
+      result += 1
+      list[result] = list[current]
+    end
+    current += 1
+  end
+  return list[0..result]
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: n, as my algorithm needs to iterate through all elements.
```

Comment:
I would say it's O(n) if the strings are small and O(m * n) where m is the length of the longest string if they're not.